src/processor/execution_context.cpp

```diff
@@ -0,0 +1,61 @@
+#include "processor/execution_context.h"
+
+#include "processor/operator/persistent/reader/csv/csv_error.h"
+#include "processor/result/factorized_table_util.h"
+#include "processor/warning_schema.h"
+
+namespace kuzu {
+using namespace common;
+namespace processor {
+
+ExecutionContext::ExecutionContext(common::Profiler* profiler, main::ClientContext* clientContext,
+    uint64_t queryID)
+    : queryID{queryID}, profiler{profiler}, clientContext{clientContext} {
+    FactorizedTableSchema tableSchema;
+
+    const auto dataTypes = WarningSchema::getColumnDataTypes();
+    for (idx_t i = 0; i < WarningSchema::getNumColumns(); ++i) {
+        tableSchema.appendColumn(
+            ColumnSchema{false, i, LogicalTypeUtils::getRowLayoutSize(dataTypes[i])});
+    }
+
+    warningTable = std::make_unique<FactorizedTable>(this->clientContext->getMemoryManager(),
+        std::move(tableSchema));
+}
+
+// marking this function with const would be misleading here
+// NOLINTNEXTLINE(readability-make-member-function-const)
+void ExecutionContext::setWarningMessages(const std::vector<PopulatedCSVError>& messages,
+    uint64_t messageLimit) {
+    common::UniqLock lock{mtx};
+    warningLimit = messageLimit;
+
+    for (const auto& error : messages) {
```

Comment:
I would rework this for loop to reuse a set ValueVectors, and populate 2048 warnings per time to append into fTable.